### PR TITLE
[artifactory] Add the option to change the access-admin password

### LINKED ISSUE #252: Artifactory admin-user support for bootstrap.creds
**Is this a request for help?**: Yes

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Feature Request

**Version of Helm and Kubernetes**: 1.13.2

**Which chart**: Artifactory

**What happened**: 

Would like to add support for bootstrap-creds

https://jfrog.com/knowledge-base/how-to-change-the-default-password-for-access-admin-user

**What you expected to happen**: N/A

**How to reproduce it** (as minimally and precisely as possible): N/A

**Anything else we need to know**: N/A

### LINKED ISSUE #253: support existingclaim for artifactory-ha again
**Is this a request for help?**:
No. I can submit PR to put back in if necessary.
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT 

**Version of Helm and Kubernetes**:
Helm: 0.11.11

**Which chart**:
stable/artifactory-ha

**What happened**:
between 0.11.3 and 0.11.11 the existing claim concept was removed.

**What you expected to happen**:
This should remain in addition to the dynamic creation options.

Our firm does not allow us to dynamically create NFS so we have to pre-provision them and then pass that into the kubernetes manifest.

**How to reproduce it** (as minimally and precisely as possible):
Look at the templates (primary, member and nginx) all refer to something like:
`{{- if not .Values.artifactory.primary.persistence.existingClaim }}`

However it doesn't actually do anything with or account for when existingClaim has a value.

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by danielezer**
…frog.com/knowledge-base/how-to-change-the-default-password-for-access-admin-user/

#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:
Adds the option to change the default access-admin password

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
fixes #252

### CODE REVIEW

**Comment by danielezer** [comment]
Sure, thanks @eldad this is still work on progress

**Comment by danielezer** [comment]
@eldada renamed the file. This is no longer work in progress.

File: stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml [hunk 1]
```diff
@@ -234,6 +237,13 @@ spec:
           mountPath: "/artifactory_extra_conf/artifactory.cluster.license"
           subPath: {{ .Values.artifactory.license.dataKey }}
       {{- end }}
+        - name: access-bootstrap-creds
+          mountPath: "/tmp/bootstrap.creds"
```
Comment (eldada) [inline stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml]: Can you put it under `/tmp/access/`?

File: stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml [hunk 2]
```diff
@@ -88,6 +88,9 @@ spec:
           echo "Copying bootstrap configs";
           cp -Lrf /bootstrap/* /artifactory_extra_conf/;
           {{- end }}
+          mkdir -p {{ .Values.artifactory.persistence.mountPath }}/access/etc;
+          cp -Lrf /tmp/bootstrap.creds {{ .Values.artifactory.persistence.mountPath }}/access/etc/bootstrap.creds;
```
Comment (eldada) [inline stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml]: Take it from `/tmp/access/bootstrap.creds`?

File: stable/artifactory/templates/artifactory-statefulset.yaml [hunk 3]
```diff
@@ -222,6 +226,13 @@ spec:
           mountPath: "/artifactory_extra_conf/artifactory.lic"
           subPath: {{ .Values.artifactory.license.dataKey }}
       {{- end }}
+        - name: access-bootstrap-creds
+          mountPath: "/tmp/bootstrap.creds"
```
Comment (eldada) [inline stable/artifactory/templates/artifactory-statefulset.yaml]: Can you put it under `/tmp/access/`?

File: stable/artifactory/templates/artifactory-statefulset.yaml [hunk 4]
```diff
@@ -93,6 +93,9 @@ spec:
           echo "Copying bootstrap configs";
           cp -Lrf /bootstrap/* /artifactory_extra_conf/;
           {{- end }}
+          mkdir -p {{ .Values.artifactory.persistence.mountPath }}/access/etc;
+          cp -Lrf /tmp/bootstrap.creds {{ .Values.artifactory.persistence.mountPath }}/access/etc/bootstrap.creds;
```
Comment (eldada) [inline stable/artifactory/templates/artifactory-statefulset.yaml]: Take it from `/tmp/access/bootstrap.creds`?

File: stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml (hunk 1 above)
Comment (danielezer) [reply stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml]: Done.

File: stable/artifactory/templates/artifactory-statefulset.yaml (hunk 3 above)
Comment (danielezer) [reply stable/artifactory/templates/artifactory-statefulset.yaml]: Done

File: stable/artifactory/templates/artifactory-statefulset.yaml (hunk 4 above)
Comment (danielezer) [reply stable/artifactory/templates/artifactory-statefulset.yaml]: Done

File: stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml (hunk 2 above)
Comment (danielezer) [reply stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml]: Done

**Comment by danielezer** [comment]
@eldada what do you think about using the same secret for other access stuff? Instead of having just one key for bootstrap.creds we can use it for anything that you want to have in the access/etc directory. I think that could me more robust. My only problem is that the other files might not be "secrets", they could just be configuration files. WDYT?

**Comment by danielezer** [comment]
@jainishshah17 can you please take a look?

**Pull request merged**